lib/medusa/producer/rabbitmq.ex

```diff
@@ -33,13 +34,18 @@ defmodule Medusa.Producer.RabbitMQ do
     get_next_event(%{state | demand: demand})
   end
 
-  def handle_subscribe(:consumer, _opts, _from, state) do
-    {:automatic, state}
+  def handle_subscribe(:consumer, _opts, from, %{consumers: consumers} = state) do
+    new_consumers = MapSet.put(consumers, from)
+    {:automatic, %{state | consumers: new_consumers}}
   end
 
-  def handle_cancel({:down, :normal}, _from, state) do
+  def handle_cancel({:down, :normal}, from, %{consumers: consumers} = state) do
```

Comment:
is this only place where consumer needs to be removed? as there is another `handle_cancel` function